### PR TITLE
Release

### PR DESCRIPTION
**Pull request by sagararyal**
Release Notes
- [x] Failed donations can now be retried with a simple retry button, without the donor having to create a new donation.
- [x] Upgrade NPM
- [x] Fix issue where direct donations using payment links showed incorrect amount for the user.
- [x] improved currency visibility, and removed rounding off from display
- [x] handle SCA for recurrent payments
- [x] implement Recurrent payments
- [x] Improved paymentOptions
- [x] Merge Refactoring Payments PR (Funding donations broken because of quantity issue)
- [ ] ~Merge Embeddable Donations (Google pay and Apple Pay is broken)~ postponed

Tests all kind of donations with all permutations of the following:
- different currencies
- different countries
- choosing a given option for donation amount vs. enter a custom donation amount
- use recurrency once vs. monthly or yearly
- using all available payment methods: Credit Card, PayPal, Sofort, Giropay, Offline, Google Pay & Apple Pay
- logged in / logged out

Above tests then must be applied to all kind of donations
- tree donations (latest build by this link)
- bouquet donations (append ?to=proj_6x3GTD5cMRv0OeQAiIlJZ0Au to link)
- funding donations (append ?to=proj_bFH0BU0Qw02RuetpQlLOMVYX to link)

### CODE REVIEW

**Comment by norbertschuler** [comment]
- [ ] This bug for https://donate-with-planet-98m7x5otf-planetapp.vercel.app/?to=proj_6x3GTD5cMRv0OeQAiIlJZ0Au should be fixed before the release - do we have a Notion task for this someone is working on?

Bildschirmfoto 2021-12-06 um 19 05 27

File: src/Donations/Micros/PaymentStatus/FailedDonation.tsx [hunk 1]
```diff
@@ -1,40 +1,145 @@
...
+    setcurrency,
+    setshowErrorCard,
+    setpaymentSetup,
+    country,
+    paymentError,
+    setPaymentError,
+    setAmount,
   } = React.useContext(QueryParamContext);
+  const router = useRouter();
+  console.log(paymentError, "paymentError");
```
Comment (sagararyal) [inline src/Donations/Micros/PaymentStatus/FailedDonation.tsx:36]: console.log

File: src/Donations/Micros/PaymentStatus/FailedDonation.tsx [hunk 2]
```diff
@@ -1,40 +1,145 @@
...
+        setpaymentSetup(paymentSetupData.data);
+      }
+      setIsPaymentOptionsLoading(false);
+      // setdonationStep(3);
+    } catch (err) {
+      // console.log(err);
+    }
+  }
+  async function getDonation() {
+    console.log(donation, "Donation");
```
Comment (sagararyal) [inline src/Donations/Micros/PaymentStatus/FailedDonation.tsx:65]: console.log

File: src/Donations/Micros/PaymentStatus/SuccessfulDonation.tsx [hunk 3]
```diff
@@ -57,6 +70,15 @@ function SuccessfulDonation({ donation, sendToReturn }: any) {
         sendRef={sendRef}
         donor={donation.donor}
       />
+      {donationID && isMobile && (router.query.step === "thankyou") && (
+        <a
+          href={`${process.env.APP_URL}/?context=${donationID}&tenant=${tenant}`}
+          className="donations-transaction-details donation-transaction-phone-view"
+          data-test-id="referenceDonation"
+        >
+          {`Ref - ${donationID}`}
```
Comment (sagararyal) [inline src/Donations/Micros/PaymentStatus/SuccessfulDonation.tsx:79]: Change to `Txn id -`

File: src/Donations/Micros/PaymentStatus/TransferDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+              <CopyButton copiedText={copiedText} buttonFor={"bank"} t={t} />
+            </div>
+          </div>
+        </div>
+      </div>
+      <div
+        className={"mt-20 mb-20 text-center"}
+        style={{ fontStyle: "italic" }}
+      >
+        {t("common:donationRef")} {donationID}
```
Comment (sagararyal) [inline src/Donations/Micros/PaymentStatus/TransferDetails.tsx:110]: Change to common:transactionID = `Transaction id`

File: src/Layout/QueryParamContext.tsx [hunk 5]
```diff
@@ -213,7 +225,9 @@ export default function QueryParamProvider({ children }: any) {
         if (!country) {
           setcountry(paymentSetupData.data.effectiveCountry);
         }
+
         setpaymentSetup(paymentSetupData.data);
+        console.log(paymentSetupData.data, "paymentSetupData.data");
```
Comment (sagararyal) [inline src/Layout/QueryParamContext.tsx:230]: console

File: src/Utils/getImageURL.ts [hunk 6]
```diff
@@ -2,15 +2,15 @@
...
+  return `${process.env.CDN_URL}/media/cache/${category}/${variant}/${imageName}`;
 }
 
-export function getPDFFile(
-  category: string,
-  fileName: any,
-) {
+export function getPDFFile(category: string, fileName: any) {
   return `${process.env.CDN_URL}/media/uploads/pdfs/${category}/${fileName}`;
+}
```
Comment (sagararyal) [inline src/Utils/getImageURL.ts:12]: unused code. I don't think we use pdf's in donate.plant..

File: next.config.js [hunk 7]
```diff
@@ -116,6 +116,7 @@ module.exports = withPlugins([[withBundleAnalyzer], [withSourceMaps]], {
     SCHEME: scheme,
     API_ENDPOINT: `${scheme}://${process.env.API_ENDPOINT}`,
     CDN_URL: `${scheme}://${process.env.CDN_URL}`,
+    IMAGE_CDN_URL: `${scheme}://${process.env.IMAGE_CDN_URL}`,
```
Comment (sagararyal) [inline next.config.js:119]: remove

**Pull request merged**